packages/dev/core/src/Shaders/ShadersInclude/pbrBlockReflection.fx

```diff
@@ -260,24 +260,33 @@
         // _____________________________ Irradiance ________________________________
         vec3 environmentIrradiance = vec3(0., 0., 0.);
 
-        #ifdef USESPHERICALFROMREFLECTIONMAP
-            #if defined(NORMAL) && defined(USESPHERICALINVERTEX)
-                environmentIrradiance = vEnvironmentIrradiance;
-            #else
+        #if defined(USESPHERICALFROMREFLECTIONMAP) || defined(USEIRRADIANCEMAP)
+            #if !defined(NORMAL) || !defined(USESPHERICALINVERTEX)

```

Comment:
I think it would be better to do:

```glsl
#if (defined(USESPHERICALFROMREFLECTIONMAP) && (!defined(NORMAL) || !defined(USESPHERICALINVERTEX))) || defined(USEIRRADIANCEMAP)
```